homeassistant/components/airpatrol/coordinator.py

```diff
@@ -0,0 +1,94 @@
+"""Data update coordinator for AirPatrol."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+from airpatrol.api import AirPatrolAPI, AirPatrolAuthenticationError, AirPatrolError
+
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN, LOGGER, SCAN_INTERVAL
+
+if TYPE_CHECKING:
+    from . import AirPatrolConfigEntry
+
+
+class AirPatrolDataUpdateCoordinator(DataUpdateCoordinator[list[dict[str, Any]]]):
+    """Class to manage fetching AirPatrol data."""
+
+    config_entry: AirPatrolConfigEntry
+    api: AirPatrolAPI | None = None
```

Comment:
```suggestion
    api: AirPatrolAPI
```
Use `self._async_setup()` in the coordinator to initialize the api. Then we don't need to get it on every refresh.